services/src/northbound/src/main/java/org/openkilda/northbound/service/impl/FlowServiceImpl.java

```diff
@@ -805,6 +825,41 @@ public FlowCacheSyncResults syncFlowCache(SynchronizeCacheAction syncCacheAction
         return response.getPayload();
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @SuppressWarnings("squid:S1192")
+    public void validateSwitches(String correlationId, List<SwitchInfoData> data,
+                                 SwitchId sourceId, SwitchId destinationId) {
+        List<SwitchInfoData> switches = data.stream().filter(switchData ->
+                Objects.equals(switchData.getSwitchId(), sourceId)
+                        || Objects.equals(switchData.getSwitchId(), destinationId))
+                .collect(Collectors.toList());
+        boolean source = switches.stream().anyMatch(sw -> Objects.equals(sw.getSwitchId(), sourceId));
+        boolean destination = switches.stream().anyMatch(sw -> Objects.equals(sw.getSwitchId(), destinationId));
+        final String errorType = "Could not create a flow";
+        if (!source && !destination) {
+            String errorMessage =
+                    String.format("Source switch %s and Destination switch %s are not connected to the controller",
+                            sourceId, destinationId);
+            logger.error("{}, CorrelationId={}", errorMessage, correlationId);
```

Comment:
You don't need to manually log correlationId, it is logged aromatically via NDC.